app.js

```diff
@@ -0,0 +1,174 @@
+var notesList = document.querySelector(".notes-list");
+var noteTitle = document.querySelector(".note-title");
+var notenote = document.querySelector(".note-note");
+var noteItem = document.querySelector(".note-item");
+var notenote = document.querySelector(".note-note");
+var emptyState = document.querySelector(".empty-state");
+
+var selectedNote = 0;
+var notes = [];
+
+toggleSwitch = document.querySelector('.theme-switch input[type="checkbox"]');
+
+function switchTheme(e) {
+  if (e.target.checked) {
+    document.documentElement.setAttribute("data-theme", "dark");
+    localStorage.setItem("theme", "dark");
+  } else {
+    document.documentElement.setAttribute("data-theme", "light");
+    localStorage.setItem("theme", "light");
+  }
+}
+
+if (currentTheme) {
+  toggleSwitch.checked = true;
+}
+
+toggleSwitch.addEventListener("change", switchTheme, false);
+
+function initializeNotes() {
+  var notesArr = localStorage.getItem(["notes"]);
+  if (JSON.parse(notesArr) == null || JSON.parse(notesArr).length === 0) {
+    notes = [
+      {
+        title: "Welcome to NotesHD!",
+        note: `This is a sample note, please feel free to edit or delete this note and get started with creating your own notes!`,
+      },
+    ];
+  } else {
+    notes = JSON.parse(notesArr);
+  }
+}
+
+function getNotesCount() {
+  return Number(notes.length);
+}
+
+function isLastEmpty() {
+  if (
+    notes[getNotesCount() - 1].title === "" &&
+    notes[getNotesCount() - 1].note === ""
+  ) {
+    return true;
+  }
+  return false;
+}
+
+function titleChangeHandler(idx) {
+  var noteTitle = document.querySelector(".note-title");
```

Comment:
You have already stored this DOM element at the top. You can use that?

Reply:
Turns out that I was trying to store a few elements like this one even before they were rendered and hence they would return null when trying to store on top. Got rid of such statements on top and used them where they were required, after they had been rendered.